### PR TITLE
Config system

### PR DESCRIPTION
**Pull request by ps2016-mt**

### CODE REVIEW

File: nba/config_classes.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from pydantic import BaseModel
+
+class DataPaths(BaseModel):
+    data_file: str
+    another_file: str
```
Comment (ty338jy) [inline nba/config_classes.py]: Hey Pushpin @ps2016-mt,
I think the config_classes here doesn't match the config.yaml!
In config.yaml we have both class Data and class Model, which should be the same structure

File: nba/config.py [hunk 2]
```diff
@@ -1 +1,11 @@
-# TODO: read in your yaml file and parse it using the data classes defined in dataclasses.py
+import yaml
+from pydantic import BaseModel
+from config_classes import DataPaths
```
Comment (ty338jy) [inline nba/config.py]: Hey Pushpin @ps2016-mt , 
I think we might have a mismatch with config.yaml?
As config.yaml has data and model, config_classes should have  class Data and class Model, in config.py we should have 'from config_classes import Data, Model'

**Comment by ty338jy** [comment]
Hello, well done! Test run successful, please merge!

**Review by ty338jy — APPROVED** [review]

**Review by cameds58 — DISMISSED** [review]
looks good

**Review by yuanjingshi — CHANGES_REQUESTED** [review]
The environment yml file has a typo in its name and cannot setup environment.

**Pull request closed without merging**